warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IteratorBuildingVisitor.java

```diff
@@ -376,8 +378,18 @@ else if (source instanceof ASTNRNode)
             analyzer = new JavaRegexAnalyzer(String.valueOf(JexlASTHelper.getLiteralValue(node)));
             
             LiteralRange<String> range = new LiteralRange<>(JexlASTHelper.getIdentifier(node), NodeOperand.AND);
-            range.updateLower(analyzer.getLeadingOrTrailingLiteral(), true);
-            range.updateUpper(analyzer.getLeadingOrTrailingLiteral() + Constants.MAX_UNICODE_STRING, true);
+            if (!analyzer.isLeadingLiteral()) {
+                // if the range is a leading wildcard we have to seek over the whole range since it's forward indexed only
+                range.updateLower(Constants.NULL_BYTE_STRING, true);
+                range.updateUpper(Constants.MAX_UNICODE_STRING, true);
+            } else {
+                range.updateLower(analyzer.getLeadingOrTrailingLiteral(), true);
```

Comment:
If we know we should be using the leading literal here, use getLeadingLiteral().

Reply:
will update